common/setups/hybrid/nn_system.py

```diff
@@ -0,0 +1,167 @@
+__all__ = ["NnArgs", "NnSystem"]
+
+import sys
+from typing import Dict, Optional
+
+# -------------------- Sisyphus --------------------
+
+import sisyphus.toolkit as tk
+import sisyphus.global_settings as gs
+
+# -------------------- Recipes --------------------
+
+import i6_core.am as am
+import i6_core.corpus as corpus_recipes
+import i6_core.meta as meta
+import i6_core.rasr as rasr
+import i6_core.returnn as returnn
+import i6_core.util as util
+
+from .base_system import BaseSystem
+
+from .util import HybridInitArgs, HybridDataInput, NnArgs
+
+# -------------------- Init --------------------
+
+Path = tk.setup_path(__package__)
+
+# -------------------- System --------------------
+
+
+class NnSystem(BaseSystem):
+    """
+    - 5 corpora types: train, devtrain, cv, dev and test
+    - two training data settings: defined in returnn config or not
+    - 3 different types of decoding: returnn, rasr, rasr-wei
+    - 3 different lm: count, lstm, trafo
```

Comment:
I would only differentiate 2 kinds: count, neural

where neural includes FFNN, LSTM, Trafo, ...

(edit:) See comment below, I think this distinction can/should be dropped.